### PR TITLE
Implement Table Chart color categorizing for each individual column

### PR DESCRIPTION
**Pull request by CharukaK**
## Purpose
Current implementation doesn't allow configuration of color based categorization column by column when applied through the configuration it will be applied for the whole Table Chart like shown below

screenshot from 2018-04-18 08-17-35

```javascript
{
    "charts": [
        {
            "type": "table",
            "columns": [
                "EngineType",
                "torque",
                "rpm"
            ],
            "columnTitles": [
                "Engine Type",
                "Engine Torque",
                "Engine RPM"
            ]
        }
    ],
    "maxLength": 7,
    "colorBasedStyle": true
}
```

With this the table chart will allow users to individually configure color based categorizing for columns.

screenshot from 2018-04-18 08-20-28

```javascript
{
    "charts": [
        {
            "type": "table",
            "columns": [
                {
                    "name": "EngineType",
                    "title": "EngineType",
                    "colorBasedStyle": true
                },
                {
                    "name": "torque",
                    "title": "Engine Torque"
                },
                {
                    "name": "rpm",
                    "title": "Engine RPM",
                    "colorBasedStyle": true,
                    "colorScale": [
                        "red"
                    ]
                }
            ]
        }
    ],
    "maxLength": 7,
    "colorBasedStyle": true
}
```

## Test environment
Node.JS v8.8.1, NPM v5.4.2

### CODE REVIEW

**Comment by this** [comment]
@CharukaK Does the table chart sample showcase this feature?

File: samples/App.jsx [hunk 1]
```diff
@@ -118,8 +118,22 @@ export default class App extends React.Component {
...
+			{
+			    name: 'torque',
+			    title: 'Engine Torque',	
+			},
+			{
+			    name: 'rpm',
+			    title: 'Engine RPM',
+			},
+		    ],	
+                    //columns: ['EngineType', 'torque', 'rpm'],
```
Comment (this) [inline samples/App.jsx]: Why commented-out??

File: samples/chart-docs/TableChartSamples.jsx [hunk 2]
```diff
@@ -37,8 +37,25 @@ export default class TableChartSamples extends React.Component {
...
+                            title: 'Engine Torque',
+                        },
+                        {
+                            name: 'rpm',
+                            title: 'Engine RPM',
+                            colorBasedStyle: true,
+                            colorScale: ['red'],
+                        },
+                    ],
+                    // columns: ['EngineType', 'torque', 'rpm'],
```
Comment (this) [inline samples/chart-docs/TableChartSamples.jsx]: Why commented-out??

File: src/components/resources/css/tableChart.css [hunk 3]
```diff
@@ -478,3 +478,16 @@
...
+}
+
+.rt-td .cell-data {
+    text-align: center;
+}
+
+.darkTheme .rt-td .cell-data {
+    text-align: center;
+    color: #ffffff;
+}
```
Comment (this) [inline src/components/resources/css/tableChart.css]: Missing file-ending new line

File: src/components/resources/css/tableChart.css [hunk 4]
```diff
@@ -478,3 +478,16 @@
...
+.darkTheme .ReactTable .rt-tbody .rt-tr-group {
+    border-bottom: 1px solid #818181;
+}
+
+.rt-td .cell-data {
+    text-align: center;
+}
+
+.darkTheme .rt-td .cell-data {
+    text-align: center;
```
Comment (erangatl) [inline src/components/resources/css/tableChart.css]: No need to define text-align, it inherits from line 487

File: src/components/resources/css/tableChart.css [hunk 5]
```diff
@@ -478,3 +478,16 @@
...
+    color: #ffffff;
```
Comment (erangatl) [inline src/components/resources/css/tableChart.css]: This can optimize to #fff

**Comment by CharukaK** [comment]
@this Yes, the samples are included in the documentation

**Pull request merged**